pkg/gofr/gofr_test.go

```diff
@@ -746,3 +932,34 @@ func Test_Shutdown(t *testing.T) {
 
 	assert.Contains(t, logs, "Application shutdown complete", "Test_Shutdown Failed!")
 }
+
+func TestApp_Subscribe(t *testing.T) {
+	test := []struct {
+		name          string
+		isInitialized bool
+	}{
+		{"subscriber is initialized", true},
+		{"subscriber is not initialized", false},
+	}
+
+	for i, tt := range test {
+		t.Run(tt.name, func(t *testing.T) {
+			app := New()
+
+			if tt.isInitialized {
+				mockContainer := container.Container{
+					Logger: logging.NewLogger(logging.ERROR),
+					PubSub: mockSubscriber{},
+				}
+
+				app.container = &mockContainer
+			}
+
+			app.Subscribe("Hello", nil)
+
+			_, ok := app.subscriptionManager.subscriptions["Hello"]
+
+			assert.Equal(t, tt.isInitialized, ok, "TEST[%d], Failed.\n%s", i, tt.name)
+		})
+	}
+}
```

Comment:
As the use cases are limited I would have written this

```suggestion
func TestApp_Subscribe(t *testing.T) {

	t.Run("subscriber is initialized", func(t *testing.T) {
		app := New()

		mockContainer := container.Container{
			Logger: logging.NewLogger(logging.ERROR),
			PubSub: mockSubscriber{}}
				
		app.container = &mockContainer
			

		app.Subscribe("Hello", nil)

		_, ok := app.subscriptionManager.subscriptions["Hello"]

		assert.True(t, ok)
	})

	t.Run("subscriber is not initialized", func(t *testing.T) {
		app := New()
		app.Subscribe("Hello", nil)

		_, ok := app.subscriptionManager.subscriptions["Hello"]

		assert.False(t, ok)
	})
}
```

Reply:
Please adapt according to
https://github.com/gofr-dev/gofr/pull/920#discussion_r1709263276